### DISCUSSION TITLE
Supports time partitioned memtable

### DISCUSSION THREAD

**Discussion by evenyag** [GreptimeTeam/greptimedb#301] [Ideas]
Now we manually create the memtable for each time range when preprocessing the write request, which also require us to precompute all time ranges of the request.

Consider alter the memtable to support creating memtable automatically according to the time range of the data

**Comment by waynexia** [comment]
The partition(`time_range()`) is an interface of `WriteBatch`. The current flow looks pretty neat to me. Do you mean to move the "pre-alloc memtables" logic into `Inserter::insert_memtables()`?

**Reply by evenyag** [reply]
I'm not quite sure now, maybe done in memtable implementation since the implementation would maintain multiple sub memtable.

**Comment by waynexia** [comment]
closed via https://github.com/GreptimeTeam/greptimedb/pull/442
